Tasks/PublishSymbols/Invoke-IndexSources.ps1

```diff
@@ -501,6 +501,17 @@ try {
                 }
             }
 
+            if ((!$workspace) -and $env:BUILD_REPOSITORY_TFVC_WORKSPACE) {
+                Write-Verbose "Attempting to resolve workspace by name: $env:BUILD_REPOSITORY_TFVC_WORKSPACE"
+                try {
+                    $workspace = $versionControlServer.GetWorkspace($env:BUILD_REPOSITORY_TFVC_WORKSPACE, '.')
+                } catch [Microsoft.TeamFoundation.VersionControl.Client.WorkspaceNotFoundException] {
+                    Write-Verbose "Workspace not found."
+                } catch {
+                    Write-Verbose "Determination failed. Exception: $_"
+                }
+            }
+
             if (!$workspace) {
                 Write-Warning (Get-LocalizedString -Key 'Unable to determine workspace from source folder ''{0}''.' -ArgumentList $sourcesRootPath)
```

Comment:
I would move this write-warning to line 503, since the new section you added is try to query workspace by name.